tool/src/java/org/sakaiproject/attendance/tool/pages/ExportPage.java

```diff
@@ -115,235 +112,140 @@ public void onSubmit() {
+                            currentCell.setCellValue("LE");
                         } else {
-                            this.holder = "";
-                        }
-                        HSSFCell cell = row.createCell(cellCount[0]);
-                        if(blankSheet){
-                            cell.setCellValue("");
-                        }else{
-                            cell.setCellValue(this.holder);
+                            currentCell.setCellValue("");   //this shows up again as part of the If structure to capture cases in which blankSheet = False but there is no data in CurrentRecord.
                         }
-                        cell.setCellType(CellType.STRING);
-                        cellCount[0]++;
-                    }
-                    if(commentsOnOff) {
-                        if (true) {
-                            this.holder = String.valueOf(attendanceRecordlist.get(columnFinder).getComment());
-                            if (this.holder.equals("null")){
-                                this.holder = "";
-                            }
-                            HSSFCell cell = row.createCell(cellCount[0]);
-                            if(blankSheet){
-                                cell.setCellValue("");
+                        if(commentsOnOff){  //true = has comments. if true, add the comment data for each student.
+                            columnCounter++;
+                            currentCell = studentRow.createCell(columnCounter, CellType.STRING);
+                            if(blankSheet){ //we can just stick blank Strings in if the sheet is to be blank.
+                                currentCell.setCellValue("");
                             }else{
-                                cell.setCellValue(this.holder);
+                                currentCell.setCellValue(currentRecord.getComment());
                             }
-                            cell.setCellType(CellType.STRING);
-                            cellCount[0]++;
                         }
+                        break;  //stop iterating when the correct data is found, after it's processed
                     }
                 }
-                rowCount[0]++;
-                this.rowCounter++;
+                columnCounter++; //increment this here because this counter is not in the For loop's header.
             }
-            FileOutputStream fos = new FileOutputStream(tempFile);
-            wb.write(fos);
+        }
+        return writeFile(wb, tempFile);
+    }
 
+    private File writeFile(HSSFWorkbook wb, File tempFile){
+        FileOutputStream fos = null;
+        try {
+            fos = new FileOutputStream(tempFile);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        try {
+            wb.write(fos);
             fos.close();
             wb.close();
```

Comment:
this isn't guaranteed to close, is it?

better to use try/with/resources pattern of try/catch/finally to be sure it's closed

Reply:
I have implemented the try/catch with resources model as you described, on the next commit. Thanks!

Reply:
Should the wb.close() be in a finally instead of in the try?

Reply:
Done. This required a separate try/catch to be around the writeFile method's call just above. WriteFile can Throw an IOexception for an error in closing the Workbook resource, and it catches its own IOException on an error in closing the FileOutputStream.
These changes are in the latest commit.